Automattic/studio
# Refactor the export manager logic

Related to https://github.com/Automattic/dotcom-forge/issues/8363

## Proposed Changes

This PR refactors the export manager and removes validators. The reason for this is that validators don't validate. They were doing various things that made more sense to be closer to the exporter.

Additionally, this fixes the issue where even when you do an SQL export, it matches the default exporter and exports the zip contents to an SQL file. We now check for the requested file extension as well. If an SQL file is requested, it will match the SQL exporter, which will only export the database.


## Testing Instructions


You can run the unit tests and additionally run the following steps:
- Start studio with `export STUDIO_IMPORT_EXPORT=true`
- Go to the import/export tab of a site and try to export the full site and just the SQL file.
- A full site export should create a tar.gz file while the database export should just trigger some console logs
 


## Pre-merge Checklist
